### PR TITLE
move batchnorm and layernorm fusion to decomposition

### PR DESCRIPTION
**Pull request by wanchaol**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #20039 dispatch and expose linear op
* **#19989 move batchnorm and layernorm fusion to decomposition**
* #19988 split canonicalize_ops, make a decompose pass
* #19987 lower batchmm to non-diff optimization

### CODE REVIEW

**Comment by wanchaol** [comment]
This is closed after a discussion with @jamesr66a , the main reason is that we don't check isFusableDevice out of fuser, and this move enables the decomposition happens on cpu side as well, which seems not ideal for the reason: 1. we don't enable cpu fuser currently 2. this decomposition will make batch_norm op disappear and could not call the faster batch_norm op during inference
